### PR TITLE
VIDEO-12411 [Chrome 112+][Room.getStats()] Read from RTCMediaSourceStats instead of RTCMediaStreamTrackStats

### PR DESCRIPTION
**Pull request by manjeshbhargav**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Description

* Fix for #1968.
* Fix for #1960.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

### CODE REVIEW

**Review by PikaJoyce — APPROVED** [review]

**Review by schayapathi — APPROVED** [review]

**Review by PikaJoyce — APPROVED** [review]

**Comment by henbos** [comment]
FYI this is not "Chrome 112+". The entire track stats object is removed-by-default in M111, but even before then there is an experiment (currently 50% Stable) that removes track in M109. So consider this an M109+ change.

**Comment by anna-vasilko** [comment]
Hi @henbos ,
Just wanted to double check that the latest plan (based on [this comment](https://groups.google.com/a/chromium.org/g/blink-dev/c/RsIktnGhHqw/m/K5bMuDIzAAAJ)) is to revert the experiment, correct?
At Twilio we really hope the experiment can be reverted asap given that it is breaking for some of our customers and their end users in the wild. The adoption of our new sdk release (with this PR change) will take long time for most of the customers as they need to 1) learn about the issue 2) pick-up the new sdk version 3) roll out their applications with the new sdk version. For some customers it takes months. 
Basically the only way to really stop the breaking impact is to stop the experiment and give some notice and time to adopt the changes.

I have to acknowledge it took unusually long time for us to discover this breaking change and act on it. Holidays probably played a role here. But we will do a retrospective internally to understand the reasons better and to catch such issues sooner in the future.

**Comment by henbos** [comment]
Hi @anna-vasilko, as this was causing more trouble than I had anticipated, the experiment is being rolled back to 0% and the rollback should go live on Monday, 9AM Pacific Time. Please note that users need to restart their browser to get the new config.

Please provide input on the new [Intent to Deprecate](https://groups.google.com/a/chromium.org/g/blink-dev/c/NZVXsJQ7tV8). Is my understanding correct that the reason this breakage was not discovered in testing is that the test environments run with clean profiles that don't get the finch experiment configs? While I've not had the intent approved yet, I did propose to use a strategy of having enabled-by-default (in M111 or M112) but using the Finch config to overwrite the experiment to an appropriate percentage (0% for now). My thinking is that this would act as a "fire alarm" to any test environments that don't get finch configs, but that real users (assuming they have finch) would not get the experiment until further rollout. Would that help here or cause more trouble?

For twilio-video.js, please make sure there are no hard dependencies on stats of type "track" or "stream". A general comment on parsing any metric is that it is highly advisable never to assume a metric or stats object exist but to first check that it is not `undefined` before reading.

Lastly, I'm sorry this issue affected you.
